### DISCUSSION TITLE
Allow a service account access from any namespace

### DISCUSSION THREAD

**Discussion by maggie44** [istio/istio#53379] [Q&A]
I have an app that creates new namespaces and service accounts within those namespaces. I would like to grant those service accounts in those namespaces access to an api pod that exists permanently, without creating individual auth policies for each as I do not know the namespace names in advance. 

Ideally something like this:

cluster.local/ns/*/sa/generic-serviceaccount

I want to avoid granting access to all the namespaces and have stricter permissions by using service accounts but it seems the principle field doesn’t support wildcards. 

What would be the best way to go about this?

**Comment by howardjohn** [comment]
You should really not do this from a security perspective. This would mean that any namespace can create a SA with 'generic-serviceaccount' and access.

**Reply by maggie44** [reply]
It would indeed. But the only alternative I can think of is to provide namespaces access using a wildcard which would then allow privilege escalation by creating a namespace. In my particular policies namespace creation is permitted to more service accounts than service account creation, so it’s a step up at least. 

The ultimate goal here is as secure a way as possible to allow creation of namespaces and pods inside those namespaces that are permitted access to an api namespace and pod, without knowing the name of the new namespaces in advance. If there is a better way I just haven’t thought of it yet.

**Comment by ilrudie** [comment]
Can your application create the policy allowing the desired access to this your api when it creates the new namespace?

**Reply by maggie44** [reply]
If I grant the application access to create policies then it has an even higher level of privilege than before and could easily escalate privileges by adding policies to the API namespace.

**Reply by ilrudie** [reply]
The application which creates new namespaces is not something you think you can rely on? Why give it access to create namespaces and service accounts then?

**Reply by maggie44** [reply]
It is something I can rely on, but I am implementing principle of least privilege. I think allowing it to create istio policies would be even less secure than allowing it to create namespaces.
